internal/services/kusto/kusto_cluster_resource.go

```diff
@@ -579,6 +741,7 @@ func expandKustoClusterVNET(input []interface{}) *clusters.VirtualNetworkConfigu
 		SubnetId:                 subnetID,
 		EnginePublicIPId:         enginePublicIPID,
 		DataManagementPublicIPId: dataManagementPublicIPID,
+		State:                    pointer.To(clusters.VnetStateEnabled),
```

Comment:
Can we add some comments here to explain why we have this logic here?
```suggestion
		// 'State' is hardcoded to 'Enabled' for the 'None' pattern.
		// If the vNet block is present it is enabled, if the vNet block is removed it is disabled.
		SubnetId:                 subnetID,
		EnginePublicIPId:         enginePublicIPID,
		DataManagementPublicIPId: dataManagementPublicIPID,
		State:                    pointer.To(clusters.VnetStateEnabled),
```